syntax = "proto3";
package distributed;

import "Game.proto";
import "Player.proto";
import "Result.proto";
import "GameServer.proto";

service Server {
    rpc connectPlayer (Player) returns (Result);
    rpc disconnectPlayer (Player) returns (Result);
    rpc checkHealth (Health) returns (Result);
    rpc sendUpdate (Context) returns (Result);
    rpc createGame (Setting) returns (Response);
    rpc connectToGame (Register) returns (Response);
}